libs/s3/s3_versioning_common_test_lib.py

```diff
@@ -344,9 +342,14 @@ def upload_versions(s3_test_obj: S3TestLib, s3_ver_test_obj: S3VersioningTestLib
             ...
             }
     """
-    LOG.info("Creating bucket")
-    resp = s3_test_obj.create_bucket(bucket_name)
-    assert_utils.assert_true(resp[0], resp[1])
+    try:
+        bucket_exists, _ = s3_test_obj.head_bucket(bucket_name)
```

Comment:
bucket_exists should be set to True when head_bucket returns. If it runs into exception then you cannot make an assumption that bucket does not exists.

Reply:
Updated the code to check that bucket creation is performed only if HEAD Bucket returns a Not Found error.